player/configfiles.c

```diff
@@ -337,6 +337,15 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
     bool write_start = true;
     double pos = get_current_time(mpctx);
 
+    // Workaround pts being negative when pausing and seeking to the beginning
+    // of audio without video. This would save a negative start which would make
+    // mpv seek to the end and immediately advance to the next playlist entry.
+    // We can't make playing_audio_pts() always return positive values because
+    // it increases jitter with display-resample. ao_get_delay() should be fixed
+    // instead, as tracked in #12322.
+    if (pos < 0)
+        pos = 0;
```

Comment:
This looks like hacky workaround for another problem. Also two lines below there is check for `MP_NOPTS_VALUE` which this change will break.

Reply:
Another possible approach would be to ensure that `get_current_time` never returns negative. I'm not sure if that has any implications, but I would be surprised if any current usage of that expects negative numbers ever.

Reply:
Actually I don't understand why `playing_audio_pts` should be allowed to return negative values.

Reply:
Yeah, another possibility would be to have a separate function that does the delay calculation and stuff. I did remove the speed adjustment from it a while ago from it but we could go further.

Reply:
Moved it below for now.